### ISSUE TITLE
Custom Storybook for SvelteKit?

### ISSUE THREAD

**Issue by janosh** [sveltejs/kit#1979]
### Describe the problem

There's 2-year-old Sapper issue https://github.com/sveltejs/sapper-template/issues/120 in which @Rich-Harris mentioned the possibility of developing an in-house alternative to the dependency-heavy Storybook.

### Describe the proposed solution

Now that Sapper was replaced by SvelteKit, I'd be curious to hear if there's a new outlook on that?

### Alternatives considered

Storybook is nice but requires some modifications to work with `"type": "module"` packages and comes with a huge list of dependencies.

### Importance

would make my life easier

**Comment by kevmodrome** [comment]
There's a project called Svench by @rixo that you should check out: https://svench-docs.vercel.app/ - probably not worth spending time making yet another Storybook alternative when one already exists :)

**Comment by janosh** [comment]
Looks cool. Kit support is still WIP (https://github.com/rixo/svench/issues/29) and the readme says Svench is in PoC stage, but I agree, if this project develops as far as say, MDSveX, there's no point in creating another one.

**Comment by 7antra** [comment]
It seems that Storybook make great efforts to implement Svelte without config https://storybook.js.org/blog/storybook-for-svelte/

**Comment by janosh** [comment]
@7antra Yeah, I tried that but just setting it up took a minute and made the fans on my machine turn wild. Since coming to Svelte, I've been trying to avoid these sorts of heavy dependencies.

**Comment by benmccann** [comment]
I think that `svelte-kit package` is supposed to fulfill some of these needs. I saw Rich mentioned Storybook in the original issue for it: https://github.com/sveltejs/kit/issues/518

**Comment by janosh** [comment]
@benmccann `svelte-kit package` is great for publishing packages. But from my understanding, Storybook also has some auto-documentation and testing features which `svelte-kit package` lacks and isn't intended to receive. Correct me if I'm wrong here.

**Comment by benmccann** [comment]
I've never used Storybook (or `svelte-kit package` for that matter), but it looks to me like StoryBook DocsPage would be quite easy to replicate for `svelte-kit package` and is probably basically supported today. That being said I'm not very familiar with the feature and this is just my first reaction from a minute of searching. Feel free to point out if there's stuff it can do that'd be hard with `svelte-kit package`

We don't have a recommended testing setup for SvelteKit yet, but would like to do that. I'd hope that it could get close to the StoryBook setup as well

**Comment by janosh** [comment]
> I've never used Storybook (or svelte-kit package for that matter), but it looks to me like StoryBook DocsPage would be quite easy to replicate for svelte-kit package and is probably basically supported today.

I wasn't aware that `svelte-kit package` has anything along the lines of automatic documentation? Could you elaborate?

**Comment by benmccann** [comment]
Nevermind. I just took a closer look at what StoryBook provides and it's not really something SvelteKit does today. It would be cool to have though :smile:

**Comment by Tymek** [comment]
Maybe this can this be done with a vite plugin? Something like virtual modules - injecting a page with components?

**Comment by bluwy** [comment]
Re-implementing Storybook is not fun, I have spent a fair amount of time in the past trying to do so. There's a reason why Storybook is bulky today, e.g. component iframe isolation (leading to CSF spec), document-iframe inter-communication, plugins, and it's an app itself.

With that said, there's also https://github.com/vitebook/vitebook, but from a brief look into its internals, it doesn't (and probably won't?) implement Storybook's component iframe isolation feature. But seems fairly robust to try out otherwise.

**Comment by Tymek** [comment]
Vitebook looks promising. It's great to see new solutions in this space. I'll definitely try it out.

True component isolation in `iframe` was very valuable when we didn't have easy and robust style isolation. I personally crave for simple configuration, like directory regex, that will allow me to show components with knobs within existing app layout. Documentation utility over sandbox features.

**Comment by IanVS** [comment]
Just a heads-up, there's also a vite builder for storybook which works with svelte.  https://github.com/storybookjs/builder-vite.  

And storybook is making changes soon to avoid users having to install webpack at all, which should help a lot with dependency bloat and performance.

**Comment by benmccann** [comment]
We probably won't be re-implementing all of Storybook :smile: But we are working on making Storybook and SvelteKit work together well! So I'm going to close this in favor of https://github.com/sveltejs/kit/issues/5397. I have PRs out to make Storybook setup work better with `"type": "module"` and reduce the number of dependencies, which were the items mentioned in the issue description here
